src/modules/iasql_functions/iasql.ts

```diff
@@ -1288,9 +1339,26 @@ async function commitSync(
   } while (ranFullUpdate);
   const t7 = Date.now();
   logger.info(`${dbId} synced, total time: ${t7 - t1}ms`);
+  const out = iasqlPlanV3(toCreate, toUpdate, toReplace, toDelete);
+  if (recordsSynced > 0) {
+    const totalRecordsSynced = await MetadataRepo.incrementRecordsSynced(dbId, recordsSynced);
+    if (totalRecordsSynced) {
+      const user = await MetadataRepo.getUserFromDbId(dbId);
+      if (user) {
+        telemetry.logCommitApply(user?.id,
+        {
+          recordsSynced: totalRecordsSynced,
+        },
+        {
+          output: out.rows.toString()
+        })
+      }
+    }
+
+  }
   await context.orm.dropConn();
   context.orm = oldOrm;
-  return iasqlPlanV3(toCreate, toUpdate, toReplace, toDelete);
+  return out;
```

Comment:
Same here. Pull the telemetry `await`s out of the normal flow so it doesn't block the user.